lib/resque/scheduler/locking.rb

```diff
@@ -59,7 +60,11 @@ def master_lock
       end
 
       def supports_lua?
-        redis_master_version >= 2.5
+        redis_master_version >= Gem::Version.new('2.5')
+      end
+
+      def supports_resilient_lock?
+        redis_master_version >= Gem::Version.new('2.6.2')
```

Comment:
It looks like the extended `SET` options were available [since redis 2.6.12](https://github.com/redis/redis/issues/931#issuecomment-15764821).

Reply:
> I'm all for simplifying calls where appropriate.
> 
> In this case though, any user who previously qualified for the resilient lock (because the redis they were connected to was >= 2.5) will get "downgraded" to the basic lock when they upgrade this plugin.
> 
> I would prefer an implementation based on `SET NX/EX` to exist _along side_ the existing lua-based resilient lock index, not to replace it.

Good point and good catch on the Redis version.  Still kind of working on this, as there is some spec failures too.